src/test/java/com/fasterxml/jackson/databind/ser/CanonicalJsonMapper.java

```diff
@@ -0,0 +1,48 @@
+package com.fasterxml.jackson.databind.ser;
+
+import com.fasterxml.jackson.core.StreamWriteFeature;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+
+public class CanonicalJsonMapper { // TODO It would be great if we could extend JsonMapper but the return type of builder() is incompatible
```

Comment:
That's not going to be possible since with 3.0 especially there are be per-format sub-classes of `ObjectMapper` (2.x actually has these too, but with 3.0 it is more fundamentally so, to match particular `TokenStreamFactory` with matching `ObjectMapper` subtype).

Reply:
Hmmm... I feel that the implementation of the builders as is leads to a lot of code duplication. Any ideas how to improve this?